src/login/logind-session-device.c

```diff
@@ -493,3 +501,35 @@ unsigned int session_device_try_pause_all(Session *s) {
 
         return num_pending;
 }
+
+int session_device_save(SessionDevice *sd) {
+        _cleanup_free_ char *state = NULL;
+        int r;
+
+        assert(sd);
+
+        /* Store device fd in PID1, which will send it back to us on
+         * restart so revocation will continue to work. To make things
+         * simple, we send all type of devices even if they don't have
+         * the revocation mechanism so we don't have to handle them
+         * differently later.
+         *
+         * Note: for devices supporting revocation, PID1 will drop the
+         * passed fds automatically as soon as the corresponding device
+         * are revoked. */
+        r = asprintf(&state, "FDSTORE=1\nFDNAME=%s", sd->session->id);
```

Comment:
also, could you please prefix the FDNAME with something? just to be safe, in case we later on want to push other stuff to PID 1 too, and we should better not collide with the naming here. Hence, please use `FDNAME=session-%s` rather than just `FDNAME=%s` or so